### PR TITLE
support GRU export to ONNX

### PR DESCRIPTION
**Pull request by anderspapitto**

### CODE REVIEW

**Comment by anderspapitto** [comment]
(Do not merge yet. The code is complete and reviewable but tests in the onnx-fb-universe repo are upcoming.)

File: torch/nn/_functions/rnn.py [hunk 1]
```diff
@@ -391,6 +395,43 @@ def reform(x):
     return torch.onnx.symbolic_override(symbolic)
 
 
+def GRU_symbolic_builder(input_size, hidden_size, num_layers, batch_first, dropout, bidirectional, **kwargs):
```
Comment (apaszke) [inline torch/nn/_functions/rnn.py]: Can we put those somewhere in ONNX dir? This is the code that implements RNNs and it's better to not mix them up

File: torch/onnx/symbolic.py [hunk 2]
```diff
@@ -471,3 +473,90 @@ def exp(g, self):
...
+        assert False, cell_type
+
+
+def reform_weights(g, w, n, intervals):
+    slices = [g.op('Slice', w, axes_i=[0], starts_i=[x * n], ends_i=[y * n]) for x, y in intervals]
+    return g.op('Concat', *slices, axis_i=0)
+
+
+def LSTM_symbolic_builder(input_size, hidden_size, num_layers, batch_first, dropout, bidirectional, **kwargs):
+    assert not batch_first
```
Comment (houseroad) [inline torch/onnx/symbolic.py]: Here also better use _unimplemented.

**Comment by dzhulgakov** [comment]
Also please fix lint (as the CI is failing). Structure wise looks good (though the circular dependencies are starting to get messy)

**Comment by ezyang** [comment]
Lint still broken.

Note that `flake8` on the box your dev'ing on is a special snowflake and doesn't actually apply the correct rules. I often `pip install flake8` to get a more sane copy.

**Comment by anderspapitto** [comment]
any more comments? I think this is ready to ship

**Comment by soumith** [comment]
thanks @anderspapitto

**Pull request merged**